### PR TITLE
Category filters for PS1, PS2 & PSP games

### PR DESCRIPTION
**Pull request by MSuih**
As requested by #5378, this adds category filters for PS1, PS2 and PSP games so that users can hide them from the game view. Since this change leaves the `Non_Disc_Game` internal category with just HDD games, it's been renamed to `HDD_Game` and redirected to refer to that category type directly.

### CODE REVIEW

**Comment by Megamouse** [comment]
Why have the submenu? it looks kinda awkward to me :)

**Comment by MSuih** [comment]
I thought it'd make more sense to have them separated from the PS3 games. But you're right, having them in the same menu entry makes it look much cleaner, especially with the way Qt handles sizing.

**Comment by Megamouse** [comment]
Yeah it looks better now. I can't really test it though since i have no ps1 games or others

File: rpcs3/rpcs3qt/game_list_frame.h [hunk 1]
```diff
@@ -96,7 +99,8 @@ namespace category // (see PARAM.SFO in psdevwiki.com) TODO: Disc Categories
 		{ "MS", psp_save }  // data
```
Comment (Megamouse) [inline rpcs3/rpcs3qt/game_list_frame.h]: I'm adding this comment here a few lines too far down, but there are comments for "non_disc_games" in cat_boot.
You should change those accordingly

**Comment by Megamouse** [comment]
I remember now. the stuff was mostly managed in disc and non-disc because of the late toolbar that contained two buttons for both filters. I forgot to add the filters back in when I removed the toolbar.

**Comment by 13xforever** [comment]
works as advertised 
image

**Pull request merged**